src/home/invite_screen.rs

```diff
@@ -287,20 +275,10 @@ impl Widget for InviteScreen {
         if let Event::Signal = event {
             if let (false, Some(room_id), true) = (self.is_loaded, &self.room_id, cx.has_global::<RoomsListRef>()) {
                 let rooms_list_ref = cx.get_global::<RoomsListRef>();
-                let restore_status_label = self.view.label(id!(restore_status_label));
                 if !rooms_list_ref.is_room_loaded(room_id) {
-                    let status_text = if rooms_list_ref.all_known_rooms_loaded() {
-                        self.all_rooms_loaded = true;
-                        format!(
-                            "An invite to room \"{}\" was not found in the homeserver's list of all rooms.\n\n\
-                             You may close this screen.",
-                            self.room_name.as_deref().unwrap_or_else(|| room_id.as_str())
-                        )
-                    } else {
-                        String::from("[Placeholder for Loading Spinner]\n\
-                         Waiting for this room to be loaded from the homeserver")
-                    };
-                    restore_status_label.set_text(cx, &status_text);
+                    self.all_rooms_loaded = rooms_list_ref.all_known_rooms_loaded();
+                    let restore_status_view = self.view.restore_status_view(id!(restore_status_view));
+                    restore_status_view.set_content(cx, self.all_rooms_loaded, &self.room_name);
```

Comment:
I noticed that you also call `set_content()` within the draw function (L394). 

If you're calling it there, why do we also need to call it here? Seems redundant. I think you can just do something like this:

```suggestion
                    self.all_rooms_loaded = rooms_list_ref.all_known_rooms_loaded();
                    self.redraw(cx);
```